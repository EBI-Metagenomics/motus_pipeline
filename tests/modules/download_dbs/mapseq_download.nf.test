nextflow_process {
    
    name "Test module download mapseq db"
    script "modules/mapseq.nf"
    process "GET_MAPSEQ_DB"

    test("download mapseq LSU db") {

        when {
            process {
                """
                input[0] = channel.value(params.silva_lsu_db_name)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.db).match()
        }
    }
    
    
    test("download mapseq SSU db") {
        when {
            process {
                """
                input[0] = channel.value(params.silva_ssu_db_name)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.db).match()
        }
    }
}