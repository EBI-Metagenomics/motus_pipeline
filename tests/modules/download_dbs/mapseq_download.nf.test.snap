{
    "download mapseq LSU db": {
        "content": [
            [
                [
                    "LSU.fasta:md5,96dc05a4ab2933ec6f443a1a0ce0d225",
                    "LSU.fasta.mscluster:md5,19f454e601925abbde17302ddc20c165",
                    "lsu2.otu:md5,515df01f33ed8d5741fbe0e55e4780b9",
                    "slv_lsu_filtered2.txt:md5,d521ce911cb9b94f4d304286ed9f8c3e"
                ]
            ]
        ],
        "timestamp": "2023-03-21T16:21:15+0000"
    }
}